extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -97,6 +131,8 @@ void build(SmallRyeHealthTemplate template, RecorderContext recorder,
 
     @BuildStep
     public void kubernetes(BuildProducer<KubernetesHealthPathBuildItem> healthPathItemProducer) {
-        healthPathItemProducer.produce(new KubernetesHealthPathBuildItem(health.path));
+        healthPathItemProducer.produce(new KubernetesHealthPathBuildItem(health.healthPath));
+        healthPathItemProducer.produce(new KubernetesHealthPathBuildItem(health.livenessPath));
+        healthPathItemProducer.produce(new KubernetesHealthPathBuildItem(health.readinessPath));
```

Comment:
This part is not right.

You will need 2 different build items for the liveness and readiness probes and make use of them here:
https://github.com/quarkusio/quarkus/blob/master/extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java#L67
and here:
https://github.com/quarkusio/quarkus/blob/master/extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java#L118

/cc @geoand

Reply:
@gsmet :+1: ! Indeed, when we added the initial Kubernetes Health support, there was only one thing to use. Now that this has changed, we should have separate build items that feed the appropriate properties.

@antoinesd If you don't feel like dealing with the Kubernetes stuff in this PR, you can remove those changes and I can deal with them in a subsequent PR

Reply:
Ok will add. But should I revert what I did here ?

Reply:
Yes, those lines would need to removed and a couple new build items added (KubernetesLivenessPathBuildItem and KubernetesReadinessPathBuildItem) I guess. 
Should we be keeping the "legancy" `KubernetesHealthPathBuildItem` or not?

Reply:
I suggest to cooperate on this PR and not create a new one for the kubernetes stuff.

Reply:
OK, sounds like a plan.

Reply:
@antoinesd Looking at the health stuff, it seems like the user will be able to select whethe they choose simple `health` or the new `readiness` and `liveliness`, so I suggest that we keep `KubernetesHealthPathBuildItem` build only feed its data into the generation process when the others have not been set. WDYT?

Reply:
@geoand to be sure, do you mean that even previous code should be removed?

Reply:
@antoinesd No no :). I am just saying that you should remove these lines you have added here add something like:

```
        healthPathItemProducer.produce(new KubernetesHealthPathBuildItem(health.healthPath));
        livenessPathItemProducer.produce(new KubernetesLivenessPathBuildItem(health.livenessPath));
        readinessPathItemProducer.produce(new KubernetesReadinessPathBuildItem(health.readinessPath));
```

where the `KubernetesLivenessPathBuildItem` and `KubernetesReadinessPathBuildItem` are new build items. 

The of course changes will be needed at: https://github.com/quarkusio/quarkus/blob/master/extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java#L121 and https://github.com/quarkusio/quarkus/blob/master/extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java#L67 to use these values (and also take into account if they are missing and only `health` is set)

Reply:
I would have named them `KubernetesHealthLivenessPathBuildItem` and `KubernetesHealthReadinessPathBuildItem` with `Health` in the name.

@geoand what I'm wondering is how you will choose between the old one and the new ones?

Reply:
I am thinking that the legacy value will only be used if the new - specific value is not used.

I'm cool with the name you propose, whatever you think is best

Reply:
But you have default values for all of them. So they will always be defined AFAICS?

Reply:
Ah yes... Now I see what you are saying. In that case I guess we should just use the liveness and readiness stuff and ditch health entirely for Kubernetes. WDYT?

Reply:
Yes, my thought exactly.

Reply:
OK, lets do that then

Reply:
I would say that this part has been fixed now.